### PR TITLE
Update SmartCachePatchWSIDataset and add unittests

### PR DESCRIPTION
**Pull request by behxyz**
### Description
Turn shuffling off for `SmartCachePatchWSIDataset`, since `PatchWSIDataset` object does not support item assignment, and added unittests for `SmartCachePatchWSIDataset`.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by Nic-Ma — APPROVED** [review]
Thanks for your quick update.
Looks good to me.
Could you please also add a test case to run your `SmartCachePatchWSIDataset` with `SmartCacheHandler` to make sure it can work well with our workflows?

Thanks.

**Comment by behxyz** [comment]
> Thanks for your quick update.
> Looks good to me.
> Could you please also add a test case to run your `SmartCachePatchWSIDataset` with `SmartCacheHandler` to make sure it can work well with our workflows?
> 
> Thanks.

Thanks @Nic-Ma ,
Using `SmartCacheHandler` requires to use a MONAI engine like SupervisedTraining, right? Is there any simpler way to test it?
I have already tested it with pathology MMAR and it worked.

File: tests/test_smartcache_patch_wsi_dataset.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        for _ in range(num_epochs):
+            for j in range(len(dataset)):
+                samples = dataset[j]
+                n_patches = len(samples)
+                self.assert_samples_expected(samples, expected[i : i + n_patches])
+                i += n_patches
+            dataset.update_cache()
+        dataset.shutdown()
+
+    def camelyon_data_download(self, file_url):
```
Comment (wyli) [inline tests/test_smartcache_patch_wsi_dataset.py:152]: the method is implemented multiple times in the codebase, shall we move it into `tests/utils.py`? also, would be great to download into `tests/testing_data` folder

**Comment by Nic-Ma** [comment]
> > Thanks for your quick update.
> > Looks good to me.
> > Could you please also add a test case to run your `SmartCachePatchWSIDataset` with `SmartCacheHandler` to make sure it can work well with our workflows?
> > Thanks.
> 
> Thanks @Nic-Ma ,
> Using `SmartCacheHandler` requires to use a MONAI engine like SupervisedTraining, right? Is there any simpler way to test it?
> I have already tested it with pathology MMAR and it worked.

Hi @behxyz ,

You can refer to this test program: https://github.com/Project-MONAI/MONAI/blob/master/tests/test_handler_smartcache.py

Thanks.

File: tests/test_smartcache_patch_wsi_dataset.py (hunk 1 above)
Comment (behxyz) [reply tests/test_smartcache_patch_wsi_dataset.py:152]: Yeah that makes sense. I can do that.

**Comment by behxyz** [comment]
> > > Thanks for your quick update.
> > > Looks good to me.
> > > Could you please also add a test case to run your `SmartCachePatchWSIDataset` with `SmartCacheHandler` to make sure it can work well with our workflows?
> > > Thanks.
> > 
> > 
> > Thanks @Nic-Ma ,
> > Using `SmartCacheHandler` requires to use a MONAI engine like SupervisedTraining, right? Is there any simpler way to test it?
> > I have already tested it with pathology MMAR and it worked.
> 
> Hi @behxyz ,
> 
> You can refer to this test program: https://github.com/Project-MONAI/MONAI/blob/master/tests/test_handler_smartcache.py
> 
> Thanks.

@Nic-Ma, I need to finish all the other pathology components by the end of week and I've got a lot on my plate. I'll go back to it after that if you don't mind.

**Comment by Nic-Ma** [comment]
> > > > Thanks for your quick update.
> > > > Looks good to me.
> > > > Could you please also add a test case to run your `SmartCachePatchWSIDataset` with `SmartCacheHandler` to make sure it can work well with our workflows?
> > > > Thanks.
> > > 
> > > 
> > > Thanks @Nic-Ma ,
> > > Using `SmartCacheHandler` requires to use a MONAI engine like SupervisedTraining, right? Is there any simpler way to test it?
> > > I have already tested it with pathology MMAR and it worked.
> > 
> > 
> > Hi @behxyz ,
> > You can refer to this test program: https://github.com/Project-MONAI/MONAI/blob/master/tests/test_handler_smartcache.py
> > Thanks.
> 
> @Nic-Ma, I need to finish all the other pathology components by the end of week and I've got a lot on my plate. I'll go back to it after that if you don't mind.

Sure, sounds good.
Thanks.

File: tests/test_smartcache_patch_wsi_dataset.py (hunk 1 above)
Comment (wyli) [reply tests/test_smartcache_patch_wsi_dataset.py:152]: how is this resolved?

**Review by wyli — DISMISSED** [review]
could you please address https://github.com/Project-MONAI/MONAI/pull/1853/files#r602317744 in this PR?

File: tests/test_smartcache_patch_wsi_dataset.py (hunk 1 above)
Comment (behxyz) [reply tests/test_smartcache_patch_wsi_dataset.py:152]: @wyli, I'll do it later in another PR since it affects almost all the pathology related unittests.

**Comment by behxyz** [comment]
> could you please address https://github.com/Project-MONAI/MONAI/pull/1853/files#r602317744 in this PR?

@wyli, I'll do it later in another PR since it affects almost all the pathology related unittests. It's on my to-do list.
